BSP/STM32F413/Src/BSP_Fans.c

```diff
@@ -11,58 +11,70 @@ Uses Pins PC6,7 and PB14,15
 #include "stm32f4xx_rcc.h"
 
 GPIO_InitTypeDef GPIO_INIT_STRUCT; //struct used to initialize pins
-TIM_OCInitTypeDef TIMER_STRUCT; //struct used to configure timers
+TIM_OCInitTypeDef TIMER_OC_STRUCT; //struct used to configure output compare for timers
 TIM_TimeBaseInitTypeDef TIMER_INIT_STRUCT; //struct used to initialize PWM timers
 
 void BSP_Fans_Init(void){
+    //Enable TIM8 clock
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_TIM8, ENABLE);
+    RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM12, ENABLE);
+
     //Enable Port Clocks
     RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOC, ENABLE); 
-    RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE); 
-    //Configure Pins to Timers
+    RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
+
+    //Configure TIM pins by configuring corresponding GPIO pins
     GPIO_PinAFConfig(GPIOC, GPIO_PinSource6, GPIO_AF_TIM8);
     GPIO_PinAFConfig(GPIOC, GPIO_PinSource7, GPIO_AF_TIM8);
-    GPIO_PinAFConfig(GPIOB, GPIO_PinSource14, GPIO_AF_TIM8);
-    GPIO_PinAFConfig(GPIOB, GPIO_PinSource15, GPIO_AF_TIM8);
-    //All pins will be initialized postivie logic, pull up/pull down, 2Mhz, and alternate function
+    GPIO_PinAFConfig(GPIOB, GPIO_PinSource14, GPIO_AF_TIM12);
+    GPIO_PinAFConfig(GPIOB, GPIO_PinSource15, GPIO_AF_TIM12);
+    //All pins will be initialized for alternate function
+    GPIO_StructInit(&GPIO_INIT_STRUCT);
     GPIO_INIT_STRUCT.GPIO_Mode = GPIO_Mode_AF;
-    GPIO_INIT_STRUCT.GPIO_OType = GPIO_OType_PP; 
-    GPIO_INIT_STRUCT.GPIO_PuPd = GPIO_PuPd_DOWN;
-    GPIO_INIT_STRUCT.GPIO_Speed = GPIO_Speed_2MHz;
+    GPIO_INIT_STRUCT.GPIO_PuPd = GPIO_PuPd_UP;
+    GPIO_INIT_STRUCT.GPIO_Speed = GPIO_Speed_50MHz;

```

Comment:
Is it necessary to have this high of a clock speed when the actual PWM speed is in the kHz?

Reply:
I am not sure. I used 50MHz because that's what some example code I found used. I can test this